dspot/src/main/java/eu/stamp_project/automaticbuilder/maven/MavenAutomaticBuilder.java

```diff
@@ -66,6 +66,22 @@ public String compileAndBuildClasspath() {
         return this.classpath;
     }
 
+    private int computeClasspath(String... goals) {
+        final String pomPathname;
+        if (InputConfiguration.get().shouldExecuteTestsInParallel()) {
+            DSpotPOMCreator.createNewPomForComputingClassPathWithParallelExecution();
+            pomPathname = InputConfiguration.get().getAbsolutePathToProjectRoot()
+                    + DSpotPOMCreator.getParallelPOMName();
+        } else {
```

Comment:
The else branch is wrong. This method is responsible for:

1. Resolve the dependencies
2. Build a file that contains all the dependencies.
3. Compile the classes, test classes and resources to have a proper `target` folder.

In fact here, we must use the default `pom.xml` for several reasons:

1. The generated pom by `DSpot` detaches the `testCompile` goal from any phase. This is done to be able to execute the instrumented tests through maven goals during the assertion amplification process:
```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <executions>
                    <execution>
                        <id>default-testCompile</id>
                        <phase>none</phase>
                    </execution>
                </executions>
            </plugin>
```
Otherwise, maven would detect a difference between the sources and the current binary. Maven would recompile the test classes and thus we will the instrumentation making impossible to complete the assertion amplification process.

2. The users might have a specific way to build its test classes (don't remember the example but I know that encounter this issue).

I opened a new pull request with a fix.